dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/secret/ServiceIntegrationResourceTest.java

```diff
@@ -0,0 +1,311 @@
+        Assert.assertEquals(HttpStatus.SC_OK, detailedIntegrationResponse.getStatus());
+        final ResponseEntityView responseEntityView3 = (ResponseEntityView) detailedIntegrationResponse.getEntity();
+        final ServiceIntegrationDetailedView serviceIntegrationDetailedView1 = (ServiceIntegrationDetailedView) responseEntityView3.getEntity();
+        Assert.assertEquals(serviceIntegrationDetailedView1.getHost().getSiteId(),host.getIdentifier());
+
+        final Map<String, Param> paramsToDeleteMap = ImmutableMap.of(
+                "param1", Param.newParam("", false, Type.STRING, "", ""),
+                "param3", Param.newParam("", false, Type.STRING, "", "")
+        );
+
+        final SecretForm secretForm1 = new SecretForm();
+        secretForm1.setServiceKey(serviceKey);
+        secretForm1.setSiteId(host.getIdentifier());
+        secretForm1.setParams(paramsToDeleteMap);
+        final Response deleteIndividualSecretResponse = serviceIntegrationResource.deleteIndividualServiceIntegrationSecret(request, response, secretForm1);
+        Assert.assertEquals(HttpStatus.SC_OK, deleteIndividualSecretResponse.getStatus());
+
+        final Response detailedIntegrationResponseAfterDelete = serviceIntegrationResource.getDetailedServiceIntegration(request, response, serviceKey, host.getIdentifier());
+        Assert.assertEquals(HttpStatus.SC_OK, detailedIntegrationResponseAfterDelete.getStatus());
+
+        final ResponseEntityView responseEntityView4 = (ResponseEntityView) detailedIntegrationResponseAfterDelete.getEntity();
+        final ServiceIntegrationDetailedView serviceIntegrationDetailedView2 = (ServiceIntegrationDetailedView) responseEntityView4.getEntity();
+        final Map<String, Secret> secretsAfterDelete = serviceIntegrationDetailedView2.getSecrets();
+        Assert.assertTrue(secretsAfterDelete.containsKey("param2"));
+        Assert.assertFalse(secretsAfterDelete.containsKey("param1"));
+        Assert.assertFalse(secretsAfterDelete.containsKey("param3"));
+        
+    }
+
+
+    @Test
+    public void  Test_Create_service_descriptor_Then_Create_Service_Integration_Then_Delete_Service_Descriptor() throws IOException {
+        final Map<String, Param> paramMap = ImmutableMap.of(
+                "param1", Param.newParam("val-1", false, Type.STRING, "label", "hint"),
+                "param2", Param.newParam("val-2", false, Type.STRING, "label", "hint"),
+                "param3", Param.newParam("val-3", false, Type.STRING, "label", "hint")
+        );
+
+        final HttpServletRequest request = mock(HttpServletRequest.class);
+        final HttpServletResponse response = mock(HttpServletResponse.class);
+        final String serviceKey = String.format("lol_%d", System.currentTimeMillis());
+        final String fileName = String.format("%s.yml", serviceKey);
+        final InputStream inputStream = createServiceDescriptorFile(fileName, serviceKey, serviceKey,
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Ensure that resources like this InputStream object are closed after use](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4623512)